File: avalon-modules/avalon-game/avalon-game-calculator/src/main/java/io/github/lishangbu/avalon/game/calculator/growthrate/MediumSlowGrowthRateCalculator.java
package io.github.lishangbu.avalon.game.calculator.growthrate;

import org.springframework.stereotype.Service;

/// 较慢组成长速率计算器
///
/// @author lishangbu
/// @since 2026/2/25
@Service
public class MediumSlowGrowthRateCalculator extends AbstractGrowthRateCalculator {
  /// 获取较慢组成长速率的内部名称
  @Override
  protected String getGrowthRateInternalName() {
    return "medium-slow";
  }

  /// 较慢组：
  /// EXP = 1.2 * Lv^3 - 15 * Lv^2 + 100 * Lv - 140
  @Override
  protected int innerCalculateGrowthRate(int level) {
    return (6 * level * level * level) / 5 - 15 * level * level + 100 * level - 140;
  }
}

File: avalon-modules/avalon-game/avalon-game-calculator/src/main/kotlin/io/github/lishangbu/avalon/game/calculator/growthrate/MediumSlowGrowthRateCalculator.kt
package io.github.lishangbu.avalon.game.calculator.growthrate

import org.springframework.stereotype.Service

/**
 * 较慢组成长速率计算器
 *
 * @author lishangbu
 * @since 2026/2/25
 */
@Service
class MediumSlowGrowthRateCalculator : AbstractGrowthRateCalculator() {
    /** 获取成长速率内部名称 */
    override fun getGrowthRateInternalName(): String = "medium-slow"

    /** 计算较慢组在指定等级下的经验值 */
    override fun tryCalculateGrowthRate(level: Int): Int = (6 * level * level * level) / 5 - 15 * level * level + 100 * level - 140
}
